AdyenComponents/BACS Direct Debit/Scenes/Input/BACSDirectDebitInputPresenter.swift

```diff
@@ -32,35 +36,43 @@ internal class BACSDirectDebitPresenter: BACSDirectDebitPresenterProtocol {
 
     // MARK: - Initializers
 
-    internal init(view: BACSDirectDebitInputFormViewProtocol,
+    internal init(view: BACSInputFormViewProtocol,
                   router: BACSDirectDebitRouterProtocol,
-                  itemsFactory: BACSDirectDebitItemsFactoryProtocol) {
+                  tracker: BACSDirectDebitComponentTrackerProtocol,
+                  itemsFactory: BACSItemsFactoryProtocol,
+                  amount: Amount?) {
         self.view = view
         self.router = router
+        self.tracker = tracker
         self.itemsFactory = itemsFactory
-        setupItems()
-        setupView()
+        self.amount = amount
     }
 
-    // MARK: - BACSDirectDebitPresenterProtocol
+    // MARK: - BACSDirectDebitInputPresenterProtocol
 
     internal func viewDidLoad() {
-        view.setupNavigationBar()
+        tracker.sendEvent()
+        createItems()
+        setupView()
     }
 
-    @objc
-    internal func didCancel() {
-        router.cancelPayment()
+    internal func viewWillAppear() {
+        if paymentCancelled {
```

Comment:
Where should this be set to true? Couldn't see anywhere

Reply:
This logic has changed. You can take a look at it again